src/Microsoft.Identity.Client/Platforms/Mac/AuthenticationAgentNSWindowController.cs

```diff
@@ -0,0 +1,280 @@
+
+        [Export("windowWillClose:")]
+        public void WillClose(NSNotification notification)
+        {
+            NSApplication.SharedApplication.StopModal();
+
+            NSUrlProtocol.UnregisterClass(new ObjCRuntime.Class(typeof(CoreCustomUrlProtocol)));
+        }
+
+        public void Run(NSWindow callerWindow)
+        {
+            _callerWindow = callerWindow;
+
+            RunModal();
+        }
+
+        //webview only works on main runloop, not nested, so set up manual modal runloop
+        void RunModal()
+        {
+            var window = Window;
+            IntPtr session = NSApplication.SharedApplication.BeginModalSession(window);
+            NSRunResponse result = NSRunResponse.Continues;
+
+            while (result == NSRunResponse.Continues)
+            {
+                using (var pool = new NSAutoreleasePool())
+                {
+                    var nextEvent = NSApplication.SharedApplication.NextEvent(
+                        NSEventMask.AnyEvent, 
+                        NSDate.DistantFuture,
+                        NSRunLoopMode.Default, 
+                        true);
+
+                    //discard events that are for other windows, else they remain somewhat interactive
+                    if (nextEvent.Window != null && nextEvent.Window != window)
+                    {
+                        continue;
+                    }
+
+                    NSApplication.SharedApplication.SendEvent(nextEvent);
+
+                    // Run the window modally until there are no events to process
+                    result = (NSRunResponse)(long)NSApplication.SharedApplication.RunModalSession(session);
+
+                    // Give the main loop some time
+                    NSRunLoop.Current.LimitDateForMode(NSRunLoopMode.Default);
+                }
+            }
+
+            NSApplication.SharedApplication.EndModalSession(session);
+        }
+
+        //largely ported from azure-activedirectory-library-for-objc
+        //ADAuthenticationViewController.m
+        public override void LoadWindow()
+        {
+            var parentWindow = _callerWindow ?? NSApplication.SharedApplication.MainWindow;
+
+            CGRect windowRect;
+            if (parentWindow != null)
+            {
+                windowRect = parentWindow.Frame;
+            }
+            else
+            {
+                // If we didn't get a parent window then center it in the screen
+                windowRect = NSScreen.MainScreen.Frame;
+            }
+
+            // Calculate the center of the current main window so we can position our window in the center of it
+            CGRect centerRect = CenterRect(windowRect, new CGRect(0, 0, DEFAULT_WINDOW_WIDTH, DEFAULT_WINDOW_HEIGHT));
+
+            var window = new NSWindow(centerRect, NSWindowStyle.Titled | NSWindowStyle.Closable | NSWindowStyle.Resizable, NSBackingStore.Buffered, true)
+            {
+                BackgroundColor = NSColor.Red,
```

Comment:
You might want to consider changing this background colour to something less garish

```suggestion
                BackgroundColor = NSColor.WindowBackground,
```

In my testing using this colour instead also automagically responds to the system Light/Dark themes too (showing dark grey/black in the dark theme, and a light grey in the light theme)